modules/shortcodes/dailymotion.php

```diff
@@ -93,40 +119,98 @@ function dailymotion_shortcode( $atts ) {
 		$params = shortcode_new_to_old_params( $atts );
 		parse_str( $params, $atts_new );
 
-		foreach( $atts_new as $k => $v ) {
+		foreach ( $atts_new as $k => $v ) {
 			$atts[ $k ] = $v;
 		}
 	}
 
-	if ( isset( $atts['id'] ) ) {
+	if ( isset( $atts['id'] ) && ! empty( $atts['id'] ) ) {
 		$id = $atts['id'];
 	} else {
 		return '<!--Dailymotion error: bad or missing ID-->';
 	}
 
-	if ( ! empty( $content_width ) ) {
-		$width = min( 425, intval( $content_width ) );
-	} else {
-		$width = 425;
+	/*set width and height using provided parameters if any */
+	$width  = isset( $atts['width'] )  ? intval( $atts['width'] )  : 0 ;
+	$height = isset( $atts['height'] ) ? intval( $atts['height'] ) : 0 ;
+
+	if ( ! $width && ! $height ) {
+		if ( ! empty( $content_width ) ) {
+			$width = absint( $content_width );
+		} else {
+			$width = 425;
+		}
+		$height = $width / 425 * 334;
+	} elseif ( ! $height ) {
+		$height = $width / 425 * 334;
+	} elseif ( ! $width ) {
+		$width = $height / 334 * 425;
+	}
+
+	// Video URL
+	$video_url = esc_url( 'https://www.dailymotion.com/embed/video/' . $atts['id'] );
```

Comment:
@MathildeS you're doing later `esc_url` when you finally output the string in the `iframe` (which is the recommended practice 👍) so we can get rid of this to avoid calling the function twice. Actually, you could place this in the call you make later:
```php
$video_url = add_query_arg(
	$player_params,
	'https://www.dailymotion.com/embed/video/' . $id
);
```
and also use $id which by now is ok to use.